hyperapp.d.ts

```diff
@@ -23,7 +23,7 @@ export interface Component<Props> {
    *
    * @memberOf [VDOM]
   */
-  (data: Props, children: VNodeChild<{} | null>[]): VNode<Props>
+  (props: Props, children: VNodeChild<{} | null>[]): VNode<any>
```

Comment:
@Mytrill Looks good. What does `XX<YY>` mean in TypeScript? 🤔 😅

Reply:
I think it is similar to Rust where you have a vector of 32bit integers `Vec<i32>`

And in Go: `var a [10]int`

So in TS an Array (pretty much a Vector because it is a heap allocated array that has an unknown size at compile time) you would define an Array of numbers like: `Array<number>`

Reply:
@selfup Thanks! So, it means Array? 😄

Reply:
@JorgeBucaran I think it's a type parameter, or perhaps more like a higher-order type. `VNode` gets to decide what to do with `<any>` it can use that type for parameters/return values to functions it defines, or whatever it wants really.

Reply:
@JorgeBucaran it is a way to define what a certain type can contain. I was just using an `Array` as an example to narrow down the scope of examining the syntax 😛

Reply:
So like say you have a:

`Car` and a `BigTruck`

To narrow down scope, let's assume all `Car`s only run on gas and all `BigTrucks` only run on diesel.

We would say something like:

```
const myCar: Car<gas> = new Car();

const myBigTruck: BigTruck<diesel> = new BigTruck();
```

My examples might be awful but I hope this helps 😂

Reply:
This is called Generic : https://www.typescriptlang.org/docs/handbook/generics.html

This is more like C++ Template. 

In this case we give user the ability to use `Props` as its own type, this is the way we give to user to type a component, h, vnode props. Like this is possible in Vue or React.

```ts
interface Props {
  min: number
  max: number
}

const Slider: Component<Props> = (props, children) => WHATEVER
```

Here we want now props object to contain a `min`  and a `max` value both as a number and required.

Reply:
This is the correct answer for the scope of this PR ^^

🎉 

Thanks @Swizz!

Reply:
The most simple example about Generics.

We want a function that take a value and return the same : `identity`. 

```ts
function identity(arg: any): any {
    return arg;
}
```
But here we will lose the given type, so this is not really an identity function because the type is not saved.

Using generic
```ts
function identity<T>(arg: T): T {
    return arg;
}
```

Now we have an `identity` function for a `T` type that take a `T` argument and return a `T` result where `T` is a given type.

`identity<number>` is the same identity function for `number`
`identity<string>` is the same identity function for `string`
`identity<MyType>` is the same identity function for `MyType`
etc...